common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java

```diff
@@ -71,4 +109,20 @@ public int saveCheckinData(List<CheckIn> checkins) {
             .findAll(Sort.by(Direction.ASC, "submissionTimestamp")).iterator())
         .collect(Collectors.toList());
   }
-}
+
+  private byte[] hashLocationId(ByteString locationId) {
+    try {
+      return MessageDigest.getInstance("SHA-256").digest(locationId.toByteArray());
+    } catch (NoSuchAlgorithmException e) {
+      logger.warn("Could not apply SHA-256 hash on " + locationId
+          + " while storing TraceTimeIntervalWarnings");
+    }
+    return NO_HASH;
+  }
+
+  private byte[] randomHashPepper() {
```

Comment:
Would it make sense to create the random bytes just in the FakeCheckinsGenerator class? I can see that there is an almost indentical copy of this method

Reply:
Idea is to pass the pepper so that the FakeGenerator class can be testable. So the service will generate a random pepper, but if this is not passed, for resilience, the Generator also has the capability to generate a random pepper.